src/app/shared/components/addregressionregion/addregressionregion.component.ts

```diff
@@ -493,5 +548,152 @@ export class AddRegressionRegionModal implements OnInit, OnDestroy {
+    }
+    lim.variables.forEach((v, varIndex) => {
+      this.addVariable();
+      const control = <FormArray>this.newLimForm.get('variables');
+      control.controls[varIndex].get('id').setValue(v.id);
+      control.controls[varIndex].get('limitationID').setValue(v.limitationID);
+      this.unitTypes.forEach((unit,x) => {  
+        if (unit.id.toString() == v.unitTypeID.toString()){
+          control.controls[varIndex].get('unitTypeID').setValue(this.unitTypes[x].id);
+        }
+      });
+      this.variables.forEach((variable,x) => {  
+        if (variable.id.toString() == v.variableTypeID.toString()){
+          control.controls[varIndex].get('variableTypeID').setValue(this.variables[x].id);
+        }
+      });
+    });
+  }
+
+  public saveLimitation() { 
+    if (!this.newLimForm.value.id) {  //Can't edit limitations w/o regression region
+      this.newLimitation.splice(this.newLimForm.value.limIndex, 1);
+      this.newLimitation.push(this.newLimForm.value);
+      this.limitations = this.newLimitation;
+      this.cancelCreateLimitaiton();
+    } else {
+      this._settingsService
+      .putEntity(this.newLimForm.value.id, this.newLimForm.value, this.configSettings.nssBaseURL + this.configSettings.limitationsURL)
+      .subscribe((response:any) => {
+        if (!response.headers) {
+          this._toasterService.pop('info', 'Info', 'Limitations were Edited');
+        } else { 
+          this._settingsService.outputWimMessages(response); 
+        }
+        this.cancelCreateLimitaiton();
+        this.getLimitations();
+      }, error => {
+        if (this._settingsService.outputWimMessages(error)) { return; }
+        this._toasterService.pop('error', 'Error creating Limitations', error.message || error._body.message || error.statusText);
+      });  
+    }
+  }
   
-}
\ No newline at end of file
+  public createNewLimitation(limitationinfo) {
+    this._settingsService
+      .postEntity(limitationinfo, this.configSettings.nssBaseURL + this.configSettings.limitationsURL + '?rr=' + this.selectedRegRegion.id)
+      .subscribe((response:any) => {
+        if (!response.headers) {
+          this._toasterService.pop('info', 'Info', 'Limitations were Added');
+        } else { 
+          this._settingsService.outputWimMessages(response); 
+        }
+        this._loaderService.hideFullPageLoad();
+        this.cancelCreateLimitaiton();
+        this.getLimitations();
+      }, error => {
+        this._loaderService.hideFullPageLoad();
+        if (this._settingsService.outputWimMessages(error)) { return; }
+        this._toasterService.pop('error', 'Error creating Limitations', error.message || error._body.message || error.statusText);
+      }
+    );  
+  }
+
+  public formatLimitation() {
+    if (this.newLimForm.value.regressionRegionID) {
+      this.newLimitation = []; 
+      this.newLimitation.push(this.newLimForm.value);
+      this.createNewLimitation(this.newLimitation);
```

Comment:
We might be able to tie the limitations in with the reg region form instead, then we wouldn't need to worry about storing these and adding them separately. I think the add scenario component does that with the parameters and errors?